modules/EvseV2G/tools.cpp

```diff
@@ -0,0 +1,330 @@
+    struct timespec time;
+    clock_gettime(CLOCK_MONOTONIC, &time);
+    return time.tv_sec * 1000 + time.tv_nsec / 1000000;
+}
+
+double calc_physical_value(const int16_t& value, const int8_t& multiplier) {
+    return static_cast<double>(value * pow(10.0, multiplier));
+}
+
+bool range_check_int32(int32_t min, int32_t max, int32_t value) {
+    return ((value < min) || (value > max)) ? false : true;
+}
+
+bool range_check_int64(int64_t min, int64_t max, int64_t value) {
+    return ((value < min) || (value > max)) ? false : true;
+}
+
+void round_down(const char* buffer, size_t len) {
+    char* p;
+
+    p = (char*)strchr(buffer, '.');
+
+    if (!p)
+        return;
+
+    if (p - buffer > len - 2)
+        return;
+
+    if (*(p + 1) == '\0')
+        return;
+
+    *(p + 2) = '\0';
+}
+
+bool get_dir_filename(char* file_name, uint8_t file_name_len, const char* path, const char* file_name_identifier) {
+
+    file_name[0] = '\0';
+
+    if (path == NULL) {
+        dlog(DLOG_LEVEL_ERROR, "Invalid file path");
+        return false;
+    }
+    DIR* d = opendir(path); // open the path
+
+    if (d == NULL) {
+        dlog(DLOG_LEVEL_ERROR, "Unable to open file path %s", path);
+        return false;
+    }
+    struct dirent* dir; // for the directory entries
+    uint8_t file_name_identifier_len = strlen(file_name_identifier);
+    while ((dir = readdir(d)) != NULL) {
+        if (dir->d_type != DT_DIR) {
+            /* if the type is not directory*/
+            if ((strlen(dir->d_name) > (file_name_identifier_len)) && /* Plus one for the numbering */
+                (strncmp(file_name_identifier, dir->d_name, file_name_identifier_len) == 0) &&
+                (file_name_len > strlen(dir->d_name))) {
+                strncpy(file_name, dir->d_name, strlen(dir->d_name) + 1);
+                break;
+            }
+        }
+    }
+
+    closedir(d);
+
+    return (file_name[0] != '\0');
+}
+
+uint8_t get_dir_numbered_file_names(char file_names[MAX_PKI_CA_LENGTH][MAX_FILE_NAME_LENGTH], const char* path,
+                                    const char* prefix, const char* suffix, const uint8_t offset,
+                                    const uint8_t max_idx) {
+    if (path == NULL) {
+        dlog(DLOG_LEVEL_ERROR, "Invalid file path");
+        return 0;
+    }
+
+    DIR* d = opendir(path); // open the path
+
+    if (d == NULL) {
+        dlog(DLOG_LEVEL_ERROR, "Unable to open file path %s", path);
+        return 0;
+    }
+    struct dirent* dir; // for the directory entries
+    uint8_t num_of_files = 0;
+    uint8_t prefix_len = strlen(prefix);
+    uint8_t suffix_len = strlen(suffix);
+    uint8_t min_idx = max_idx; // helper value to re-sort array
+
+    while (((dir = readdir(d)) != NULL) && (num_of_files != (max_idx - offset))) {
+        if (dir->d_type != DT_DIR) {
+            /* if the type is not directory*/
+            if ((strlen(dir->d_name) > (prefix_len + suffix_len + 1)) && /* Plus one for the numbering */
```

Comment:
`if ((std::string(dir->d_name).size() > (prefix_len + suffix_len + 1)) && /* Plus one for the numbering */`